BasicOperations/Addition.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BasicOperations
+{
+    public class Addition
+    {
+        public static dynamic Sum(dynamic a, dynamic b)
+        {
+            return a + b;
+        }
+
+        public static dynamic Sum(dynamic arrayA)
+        {
+            dynamic c = 0;
+            foreach (dynamic a in arrayA)
+            {
+                c = Sum(a, c);
+            }
+            return c;
+        }
```

Comment:
It is probably better to not use dynamic types when we don't have to. This code could be more specific, which in the long run would improve on the efficiency as it will not have to go through the process of determining the variables at each stage.